source/Engine.cpp

```diff
@@ -2020,19 +2023,36 @@ void Engine::DoCollisions(Projectile &projectile)
 	}
 	else if(projectile.MissileStrength())
 	{
-		// If the projectile did not hit anything, give the anti-missile systems
-		// a chance to shoot it down.
+		// If the projectile did not hit anything, check for anti-missile
+		// systems in range.
 		for(Ship *ship : hasAntiMissile)
 			if(ship == projectile.Target() || gov->IsEnemy(ship->GetGovernment()))
-				if(ship->FireAntiMissile(projectile, visuals))
+				if(ship->IsInAntiMissileRangeOf(projectile))
 				{
-					projectile.Kill();
-					break;
+					antiMissileTargets[ship].push_back(&projectile);
 				}
 	}
 }
 
 
+// Fire anti-missile shots as appropriate for this ship, prioritizing nearby
+// projectiles over distant ones. Like DoCollisions, this function adds visuals
+// directly to the main visuals list. If this is multi-threaded in the future,
+// that will need to change.
+void Engine::DoAntiMissile(Ship *ship)
+{
+	sort(antiMissileTargets[ship].begin(), antiMissileTargets[ship].end(),
+			[ship](Projectile *i, Projectile *j) -> bool
```

Comment:
The idiomatic variable names for sorted elements are `a` and `b` :)
Should you end up keeping the sort step, please ensure you declare the comparator's arguments as `const`, and use a reference capture, e.g.
```
[&ship](const Projectile *a, const Projectile *b) -> bool
```

Reply:
I removed the sort step but will keep this in mind if I put it back in after profiling!